developer/src/kmc-package/src/kmp-compiler.ts

```diff
@@ -37,117 +40,227 @@ export default class KmpCompiler {
+      description: "Package information (JSON)"
+    });
+
+    //
+    // Add keyboard metadata
+    //
 
     if(kps.keyboards && kps.keyboards.keyboard) {
-      kmp.keyboards = arrayWrap(kps.keyboards.keyboard).map((keyboard: KpsFileKeyboard) => {
-        return { name:keyboard.name, id:keyboard.iD, version:keyboard.version, languages: kpsLanguagesToKmpLanguages(arrayWrap(keyboard.languages.language) as KpsFileLanguage[]) }
+      kmp.keyboards = this.arrayWrap(kps.keyboards.keyboard).map((keyboard: KpsFileKeyboard) => {
+        return {
+          displayFont: keyboard.displayFont ? path.basename(keyboard.displayFont) : undefined,
+          oskFont: keyboard.oSKFont ? path.basename(keyboard.oSKFont) : undefined,
+          name:keyboard.name,
+          id:keyboard.iD,
+          version:keyboard.version,
+          languages: this.kpsLanguagesToKmpLanguages(this.arrayWrap(keyboard.languages.language) as KpsFileLanguage[])
+        };
       });
     }
 
+    //
+    // Add lexical-model metadata
+    //
+
     if(kps.lexicalModels && kps.lexicalModels.lexicalModel) {
-      kmp.lexicalModels = arrayWrap(kps.lexicalModels.lexicalModel).map((model: KpsFileLexicalModel) => {
-        return { name:model.name, id:model.iD, languages: kpsLanguagesToKmpLanguages(arrayWrap(model.languages.language) as KpsFileLanguage[]) }
+      kmp.lexicalModels = this.arrayWrap(kps.lexicalModels.lexicalModel).map((model: KpsFileLexicalModel) => {
+        return { name:model.name, id:model.iD, languages: this.kpsLanguagesToKmpLanguages(this.arrayWrap(model.languages.language) as KpsFileLanguage[]) }
       });
     }
 
-    if(kps.startMenu) {
+    //
+    // Add Windows Start Menu metadata
+    //
+
+    if(kps.startMenu && kps.startMenu.items) {
       kmp.startMenu = {};
       if(kps.startMenu.addUninstallEntry) kmp.startMenu.addUninstallEntry = kps.startMenu.addUninstallEntry === '';
       if(kps.startMenu.folder) kmp.startMenu.folder = kps.startMenu.folder;
-      if(kps.startMenu.items && kps.startMenu.items.item) kmp.startMenu.items = arrayWrap(kps.startMenu.items.item);
+      if(kps.startMenu.items && kps.startMenu.items.item) kmp.startMenu.items = this.arrayWrap(kps.startMenu.items.item);
     }
 
+    //
+    // Add translation strings
+    //
+
     if(kps.strings && kps.strings.string) {
-      kmp.strings = arrayWrap(kps.strings.string);
+      kmp.strings = this.arrayWrap(kps.strings.string);
     }
 
-    //let util = require('util');
-    //console.log(util.inspect(kmp, false, null))
-    //console.log(kps);
+    kmp = this.stripUndefined(kmp) as KmpJsonFile;
 
     return kmp;
   }
 
+    // Helper functions
+
+  private kpsInfoToKmpInfo(info: KpsFileInfo): KmpJsonFileInfo {
+    let ni: KmpJsonFileInfo = {};
+
+    const keys: [(keyof KpsFileInfo), (keyof KmpJsonFileInfo)][] = [
+      ['author','author'],
+      ['copyright','copyright'],
+      ['name','name'],
+      ['version','version'],
+      ['webSite','website']
```

Comment:
capital `S` answers 'why is this not a one dimensional list'

Reply:
Yeah, gross right? It keeps me awake at night. Note that there is a camel-casify thing that I asked the xml2js wrapper to do when I wrote this code originally which I now regret and will back out in the future, which will mean more of these tables (but then at least it will say `WebSite` and `ID` just like the XML rather than this weird hybrid `webSite` and `iD` which matches nothing in any format!)

Reply:
--> tracking in #7353